RetailCoder.VBE/Inspections/ObjectVariableNotSetInspection.cs

```diff
@@ -2,62 +2,45 @@
     {
+
         public ObjectVariableNotSetInspection(RubberduckParserState state)
-            : base(state, CodeInspectionSeverity.Error)
-        {
-        }
+            : base(state, CodeInspectionSeverity.Error) {  }
 
         public override CodeInspectionType InspectionType { get { return CodeInspectionType.CodeQualityIssues; } }
 
         public override IEnumerable<IInspectionResult> GetInspectionResults()
         {
-            var interestingDeclarations =
-                State.AllUserDeclarations.Where(item =>
-                        !item.IsSelfAssigned &&
-                        !item.IsArray &&
-                        !SymbolList.ValueTypes.Contains(item.AsTypeName) &&
-                        (item.AsTypeDeclaration == null || (!ClassModuleDeclaration.HasDefaultMember(item.AsTypeDeclaration) &&
-                        item.AsTypeDeclaration.DeclarationType != DeclarationType.Enumeration &&
-                        item.AsTypeDeclaration.DeclarationType != DeclarationType.UserDefinedType)) &&
-                        (item.DeclarationType == DeclarationType.Variable ||
-                         item.DeclarationType == DeclarationType.Parameter));
-
-            var interestingMembers =
-                State.AllUserDeclarations.Where(item =>
-                    (item.DeclarationType == DeclarationType.Function || item.DeclarationType == DeclarationType.PropertyGet)
-                    && !item.IsArray
-                    && item.IsTypeSpecified
-                    && !SymbolList.ValueTypes.Contains(item.AsTypeName) 
-                    && (item.AsTypeDeclaration == null // null if unresolved (e.g. in unit tests)
-                        || (item.AsTypeDeclaration.DeclarationType != DeclarationType.Enumeration && item.AsTypeDeclaration.DeclarationType != DeclarationType.UserDefinedType 
-                            && item.AsTypeDeclaration != null 
-                            && !ClassModuleDeclaration.HasDefaultMember(item.AsTypeDeclaration))));
-
-            var interestingReferences = interestingDeclarations
-                    .Union(interestingMembers.SelectMany(item =>
-                        item.References.Where(reference => reference.ParentScoping.Equals(item) && reference.IsAssignment)
-                    .Select(reference => reference.Declaration)))
-                    .SelectMany(declaration =>
-                        declaration.References.Where(reference =>
-                        {
-                            var letStmtContext = ParserRuleContextHelper.GetParent<VBAParser.LetStmtContext>(reference.Context);
-                            return reference.IsAssignment && letStmtContext != null && letStmtContext.LET() == null;
-                        })
-                    );
-
-
-            return interestingReferences.Select(reference => new ObjectVariableNotSetInspectionResult(this, reference));
+            var allInterestingDeclarations =
```

Comment:
This is the terminology I found in the pre-existing code - so I retained it.  An 'interesting' declaration is a declaration who's references may (or may not) require an assignment using "Set".  I'm open to other terms if you prefer a different name.